configtx/orderer.go

```diff
@@ -185,6 +195,288 @@ func (o *OrdererGroup) Configuration() (Orderer, error) {
+		return fmt.Errorf("unmarshaling etcd raft metadata: %v", err)
+	}
+
+	etcdRaft.Options.ElectionTick = tick
+
+	consensusMetadata, err := marshalEtcdRaftMetadata(etcdRaft)
+	if err != nil {
+		return fmt.Errorf("marshaling etcdraft metadata for orderer type '%s': %v", orderer.ConsensusTypeEtcdRaft, err)
+	}
+
+	consensusTypeProto.Metadata = consensusMetadata
+
+	e.value.Value, err = proto.Marshal(consensusTypeProto)
+	return err
+}
+
+// SetHeartbeatTick sets the Etcdraft's heartbeat tick interval.
+func (e *EtcdRaftOptionsValue) SetHeartbeatTick(tick uint32) error {
+	consensusTypeProto := &ob.ConsensusType{}
+	err := proto.Unmarshal(e.value.Value, consensusTypeProto)
+	if err != nil {
+		return err
+	}
+
+	etcdRaft, err := unmarshalEtcdRaftMetadata(consensusTypeProto.Metadata)
+	if err != nil {
+		return fmt.Errorf("unmarshaling etcd raft metadata: %v", err)
+	}
+
+	etcdRaft.Options.HeartbeatTick = tick
+
+	consensusMetadata, err := marshalEtcdRaftMetadata(etcdRaft)
+	if err != nil {
+		return fmt.Errorf("marshaling etcdraft metadata for orderer type '%s': %v", orderer.ConsensusTypeEtcdRaft, err)
+	}
+
+	consensusTypeProto.Metadata = consensusMetadata
+
+	e.value.Value, err = proto.Marshal(consensusTypeProto)
+	return err
+}
+
+// SetMaxInflightBlocks sets the Etcdraft's max inflight blocks.
+func (e *EtcdRaftOptionsValue) SetMaxInflightBlocks(maxBlks uint32) error {
+	consensusTypeProto := &ob.ConsensusType{}
+	err := proto.Unmarshal(e.value.Value, consensusTypeProto)
+	if err != nil {
+		return err
+	}
+
+	etcdRaft, err := unmarshalEtcdRaftMetadata(consensusTypeProto.Metadata)
+	if err != nil {
+		return fmt.Errorf("unmarshaling etcd raft metadata: %v", err)
+	}
+
+	etcdRaft.Options.MaxInflightBlocks = maxBlks
+
+	consensusMetadata, err := marshalEtcdRaftMetadata(etcdRaft)
+	if err != nil {
+		return fmt.Errorf("marshaling etcdraft metadata for orderer type '%s': %v", orderer.ConsensusTypeEtcdRaft, err)
+	}
+
+	consensusTypeProto.Metadata = consensusMetadata
+
+	e.value.Value, err = proto.Marshal(consensusTypeProto)
+	return err
+}
+
+// SetSnapshotIntervalSize sets the Etcdraft's max inflight blocks.
+func (e *EtcdRaftOptionsValue) SetSnapshotIntervalSize(intervalSize uint32) error {
+	consensusTypeProto := &ob.ConsensusType{}
+	err := proto.Unmarshal(e.value.Value, consensusTypeProto)
+	if err != nil {
+		return err
+	}
+
+	etcdRaft, err := unmarshalEtcdRaftMetadata(consensusTypeProto.Metadata)
+	if err != nil {
+		return fmt.Errorf("unmarshaling etcd raft metadata: %v", err)
+	}
+
+	etcdRaft.Options.SnapshotIntervalSize = intervalSize
+
+	consensusMetadata, err := marshalEtcdRaftMetadata(etcdRaft)
+	if err != nil {
+		return fmt.Errorf("marshaling etcdraft metadata for orderer type '%s': %v", orderer.ConsensusTypeEtcdRaft, err)
+	}
+
+	consensusTypeProto.Metadata = consensusMetadata
+
+	e.value.Value, err = proto.Marshal(consensusTypeProto)
+	return err
+}
+
+// EtcdRaftOptions returns the existing orderer etcdraft options value.
```

Comment:
This should probably be moved up to either just below the BatchSize() call or above the Set functions for etcdraft options.

Reply:
`// EtcdRaftOptions returns an EtcdRaftOptionsValue that can be used to configure an etcdraft configuration's options.`